### ISSUE TITLE
Hide Storage Volumes Option

### ISSUE THREAD

**Issue by Wade821** [raspberrypi/rpi-imager#617]
Just an option to hide volumes as desired, as I have 0 interest in ever using the "Microsoft Storage Space Device" drives as media for the Raspberry Pi Imager. 

With perhaps a button to "Show Hidden Volumes" there that were manually hidden in order to unhide them.

![image](https://github.com/raspberrypi/rpi-imager/assets/18544575/6ae658ea-a479-4f2f-947b-fbc08ee3e000)

**Comment by tdewey-rpi** [comment]
Thanks for the report, @Wade821.

I think I can agree that writing to a grouped disk set like that isn't likely to be what someone wants to do, so filtering these out entirely seems like a solid first step. We can evaluate adding an option to show these in the UI at a later point.

Would that address your concern?

**Comment by Wade821** [comment]
Yes, I think that would work just fine. Filtering them out to avoid accidental selection would be ideal.

**Comment by tdewey-rpi** [comment]
Awesome, thanks for the response.

I'm waiting on a pair of drives to reproduce a Storage Space on my laptop, after which I'll be able to test locally. My initial thinking is that the `drivelist` module must be capable of filtering these in a straightforward manner, but without a reasonable test system I'm not prepared to start work.

**Comment by tdewey-rpi** [comment]
If you've got `npm` ready to go, I'd love to see the output from `devicelist` on your machine - in particular, I want to know if it's flagging that the Storage Spaces are actually virtual disks or not. https://www.npmjs.com/package/drivelist

**Comment by Wade821** [comment]
Ah, shoot. I'm on Windows primarily and it's been a long time since I played with WSL, as I'm using VMWare and I think that requires disabling the Hypervisor/'Virtual Machine Platform' stuff (and in turn the necessary components to play with WSL, since I recall having to turn that off to get VMWare to work).

I'm sure I could get that to work if really needed, but I'll have to do it later since that would take some time to reconfigure everything.

**Comment by tdewey-rpi** [comment]
@Wade821, oof. Yes, don't let my request mess up your system. I'll get to this issue once I've got the drives.

**Comment by tdewey-rpi** [comment]
My ordered drives arrived, but appear to have fallen foul of a poorly-documented requirement of Windows Storage Spaces - notably, that it requires drives that present 4K physical sectors. Unfortunately, mine present 512byte sectors, and no amount of gentle persuasion can convince it otherwise.

I've also tried creating virtual disks that would meet the criteria - but this _also_ failed because my laptop's internal drive _also_ uses 512byte sectors.

I'll try and source some other drives with 4K sectors to try this out.

**Comment by Wade821** [comment]
Maybe if you don't find something soon, I'll get around to messing with my system and report back. I've been working on my own project with Github/Drawpile server stuff and need the existing virtual machines until that's completed. Hopefully not much longer.

**Comment by twerthi** [comment]
Yes, please remove Storage Spaces drives from the selection list.  I don't want to get click happy when imaging my SD cards.

**Comment by tdewey-rpi** [comment]
Fixed in 2de6d0da3f7dabac6114a36af46bd864ad87bfa8

Storage spaces apparently carry a unique bus type identifier, which we can use to mark them as system drives.
